### PR TITLE
remove deprecated indices kwarg from peak_local_max

### PR DESCRIPTION
**Pull request by grlee77**
## Description

completes the scheduled removal of `indices` from peak_local_max. 

## Checklist

- [Docstrings for all functions](https://github.com/numpy/numpy/blob/master/doc/example.py)
- Gallery example in `./doc/examples` (new features only)
- Benchmark in `./benchmarks`, if your changes aren't covered by an
  existing benchmark
- Unit tests
- Clean style in [the spirit of PEP8](https://www.python.org/dev/peps/pep-0008/)
- Descriptive commit messages (see below)

## For reviewers

- Check that the PR title is short, concise, and will make sense 1 year
  later.
- Check that new functions are imported in corresponding `__init__.py`.
- Check that new features, API changes, and deprecations are mentioned in
      `doc/release/release_dev.rst`.

### CODE REVIEW

**Review by rfezzani — COMMENTED** [review]
Thank you @grlee77. I just found a small oversight :wink:

**Comment by grlee77** [comment]
> I just found a small oversight 

@rfezzani, did you mean to leave a suggestion here?

File: skimage/feature/peak.py [hunk 1]
```diff
@@ -187,18 +178,15 @@ def peak_local_max(image, min_distance=1, threshold_abs=None,
...
     -----
     The peak local maximum function returns the coordinates of local peaks
-    (maxima) in an image. Internally, a maximum filter is used for finding local
-    maxima. This operation dilates the original image. After comparison of the
-    dilated and original image, this function returns the coordinates or a mask
-    of the peaks where the dilated image equals the original image.
+    (maxima) in an image. Internally, a maximum filter is used for finding
+    local maxima. This operation dilates the original image. After comparison
+    of the dilated and original image, this function returns the coordinates or
+    a mask of the peaks where the dilated image equals the original image.
```
Comment (rfezzani) [inline skimage/feature/peak.py:189]: Oups, I forgot to validate my suggestion :sweat_smile: 
```suggestion
    of the dilated and original image, this function returns the coordinates
    of the peaks where the dilated image equals the original image.
```

File: skimage/feature/peak.py [hunk 2]
```diff
@@ -116,9 +117,8 @@ def _get_excluded_border_width(image, min_distance, exclude_border):
     return border_width
 
 
-@remove_arg("indices", changed_version="0.20")
 def peak_local_max(image, min_distance=1, threshold_abs=None,
-                   threshold_rel=None, exclude_border=True, indices=True,
+                   threshold_rel=None, exclude_border=True,
                    num_peaks=np.inf, footprint=None, labels=None,
                    num_peaks_per_label=np.inf, p_norm=np.inf):
     """Find peaks in an image as coordinate list or boolean mask.
```
Comment (mkcor) [inline skimage/feature/peak.py:124]:
```suggestion
    """Find peaks in an image as coordinate list.
```

File: skimage/feature/peak.py [hunk 3]
```diff
@@ -187,17 +178,14 @@ def peak_local_max(image, min_distance=1, threshold_abs=None,
     Returns
     -------
     output : ndarray or ndarray of bools
```
Comment (mkcor) [inline skimage/feature/peak.py:180]:
```suggestion
    output : ndarray
```

File: skimage/feature/peak.py [hunk 4]
```diff
@@ -187,17 +178,14 @@ def peak_local_max(image, min_distance=1, threshold_abs=None,
...
 
     Notes
     -----
     The peak local maximum function returns the coordinates of local peaks
-    (maxima) in an image. Internally, a maximum filter is used for finding local
-    maxima. This operation dilates the original image. After comparison of the
-    dilated and original image, this function returns the coordinates or a mask
+    (maxima) in an image. Internally, a maximum filter is used for finding
+    local maxima. This operation dilates the original image. After comparison
+    of the dilated and original image, this function returns the coordinates
```
Comment (mkcor) [inline skimage/feature/peak.py:188]:
```suggestion
    of the dilated and original images, this function returns the coordinates
```

File: skimage/feature/tests/test_peak.py [hunk 5]
```diff
@@ -359,28 +335,29 @@ def test_disk(self):
...
-                                         min_distance=1, threshold_rel=0,
-                                         threshold_abs=-1, indices=False,
-                                         exclude_border=False)
+        peak_idx = peak.peak_local_max(image, labels=np.ones((10, 20), int),
+                                       footprint=footprint,
+                                       min_distance=1, threshold_rel=0,
+                                       threshold_abs=-1,
+                                       exclude_border=False)
+        result = np.zeros_like(image, dtype=bool)
+        result[tuple(peak_idx.T)] = 1
```
Comment (mkcor) [inline skimage/feature/tests/test_peak.py:344]:
```suggestion
        result[tuple(peak_idx.T)] = True
```

File: skimage/feature/tests/test_peak.py [hunk 6]
```diff
@@ -359,28 +335,29 @@ def test_disk(self):
...
-        with expected_warnings(["indices argument is deprecated"]):
-            result = peak.peak_local_max(image, footprint=footprint,
-                                         threshold_abs=-1,
-                                         indices=False,
-                                         exclude_border=False)
+        peak_idx = peak.peak_local_max(image, footprint=footprint,
+                                       threshold_abs=-1,
+                                       exclude_border=False)
+        result = np.zeros_like(image, dtype=bool)
+        result[tuple(peak_idx.T)] = 1
```
Comment (mkcor) [inline skimage/feature/tests/test_peak.py:350]:
```suggestion
        result[tuple(peak_idx.T)] = True
```

**Review by alexdesiqueira — APPROVED** [review]

**Pull request merged**